backend/src/main/java/com/bakdata/conquery/models/datasets/Column.java

```diff
@@ -78,90 +64,21 @@ public String toString() {
-	 *                              part of the {@link com.bakdata.conquery.models.preproc.Preprocessed}-file
-	 * @param sharedDictionaryLocks A collection of locks used for the synchronized creation of shared dictionaries.
-	 */
-	public void createSharedDictionaryReplacement(Map<String, Dictionary> dicts, NamespaceStorage storage, Map<DictionaryId, Dictionary> out, IdMutex<DictionaryId> sharedDictionaryLocks) {
-		Preconditions.checkArgument(type.equals(MajorTypeId.STRING), "Not a STRING Column.");
-		Preconditions.checkArgument(sharedDictionary != null, "Can only be used for Shared Dictionary based Columns");
-		// If the column is based on a shared dict. We reference a new empty dictionary or the existing one
-		// but without updated entries. The entries are updated later on, see ImportJob#applyDictionaryMappings.
-
-		Dictionary sharedDict = null;
-		final DictionaryId sharedDictId = new DictionaryId(table.getDataset().getId(), getSharedDictionary());
-
-		try (IdMutex.Locked lock = sharedDictionaryLocks.acquire(sharedDictId)) {
-			sharedDict = storage.getDictionary(sharedDictId);
-			// Create dictionary if not yet present
-			if (sharedDict == null) {
-				sharedDict = new MapDictionary(table.getDataset(), getSharedDictionary());
-				storage.updateDictionary(sharedDict);
-			}
-		}
-		out.put(new DictionaryId(Dataset.PLACEHOLDER.getId(), dicts.get(getName()).getName()), sharedDict);
-	}
-
-
-	/**
-	 * See {@link Column#createSharedDictionaryReplacement(Map, NamespaceStorage, Map, IdMutex)}
-	 */
-	public void createSingleColumnDictionaryReplacement(Map<String, Dictionary> dicts, String importName, Map<DictionaryId, Dictionary> out) {
-		Preconditions.checkArgument(type.equals(MajorTypeId.STRING), "Not a STRING Column.");
-		Preconditions.checkArgument(sharedDictionary == null, "Cannot be used for Shared Dictionary based Columns.");
-
-		final Dictionary dict = dicts.get(getName());
-		final String name = computeDefaultDictionaryName(importName);
-
-		out.put(new DictionaryId(Dataset.PLACEHOLDER.getId(), dict.getName()), dict);
-
-		dict.setDataset(table.getDataset());
-		dict.setName(name);
-	}
-
-
-	private String computeDefaultDictionaryName(String importName) {
-		return String.format("%s#%s", importName, getId().toString());
-	}
-
 
 	@Override
 	public TrieSearch<FrontendValue> createTrieSearch(IndexConfig config, NamespaceStorage storage) {
 
 		final int suffixLength = isGenerateSuffixes() ? config.getSearchSuffixLength() : Integer.MAX_VALUE;
 
-		final TrieSearch<FrontendValue> search = new TrieSearch<>(suffixLength, config.getSearchSplitChars());
-
-		storage.getStorageHandler()
-			   .lookupColumnValues(storage, this)
-			   .map(value -> new FrontendValue(value, value))
-			   .onClose(() -> log.debug("DONE processing values for {}", getId()))
-			   .forEach(feValue -> search.addItem(feValue, FilterSearch.extractKeywords(feValue)));
+		// TODO send message to shards to collect data, then on response create trieSearch proper?
```

Comment:
Das hatte ich jetzt gebündelt im UpdateFilterSearchJob. Der sammelt gerade alle Columns zusammen und schick sie zusammen ab.

Die Ergebnisse der Columsn kommen aber einzeln zurück.

Was ist dir lieber? Eine Bulkanfrage für die Columns oder jede Column einzeln anfragen?

Reply:
Zurück schicken solltest du auf jeden Fall einzeln, dass das nicht zu groß werden. Anfragen würde ich in Bulk, dass du das in einem ThreadPool parallelisieren kannst.

Reply:
Gut dann kann der Code hier weg